### DISCUSSION TITLE
Connecting to Oracle database with IP Address, dbname

### DISCUSSION THREAD

**Discussion by ranganathanrajeshkumar** [vrogier/ocilib#356] [Q&A] [answered]
Hello Team,

I am using this lib for self-learning. I went through the demo and source code. I am still unable to find how to connect to the Oracle database with 'IP Address'. Can the community please guide how I can do that?

**Comment by vrogier** [comment]
Hi,

You can use an "easy connect" connection string:  {listener_ip}/{listener_port}:{service_name}

Regards,

Vincent

**Reply by ranganathanrajeshkumar** [reply]
inline Connection::Connection(const ostring& db, const ostring& user, const ostring& pwd, Environment::SessionFlags sessionFlags)

do we need to pass it on db parameter?, Can you please share some examples?

**Reply by vrogier** [reply, marked as answer]
You pass that easy connect string in the 'db' parameter.

For example:

Connection con("192.168.0.45:1521/db21c", "usr", "pwd"):

You must know:
- the IP/hostname where the oracle listener is
- the port used by the oracle listener
- the oracle database service name
